flang/include/flang/Optimizer/Support/InternalNames.h

```diff
@@ -137,6 +137,11 @@ struct NameUniquer {
   static bool belongsToModule(llvm::StringRef uniquedName,
                               llvm::StringRef moduleName);
 
+  /// Given a mangled derived type name, get the name of the related derived
+  /// type descriptor object. Returns an empty string if \p mangledTypeName is
+  /// not a valid mangled derived type name.
+  static std::string getTypeDescriptorName(llvm::StringRef mangledTypeName);
```

Comment:
The internal name is now harmonious through the compiler with recent changes to the front end, correct?